docs/DeveloperGuide.md

```diff
@@ -1,29 +1,194 @@
+- Upon invoking the `set-genre` command, `ParserMain` interprets the command and forwards it to `ParserGenre`.
+ `ParserGenre` then guides the user through selecting an existing genre or adding a new one, ensuring valid inputs at 
+  each step.
+- The chosen genre is then applied to the specified book through `BookGenre`, which interfaces with `BookList` to make 
+  the update.
+- Throughout this process, `Ui` is called upon to display messages, guiding the user and confirming the successful 
+  update.
+
+#### Rationale for Design:
+
+- The command pattern used allows for easy addition of new commands and features without altering existing code 
+  structures, adhering to open/closed principles.
+- The clear separation of concerns makes the system robust, with each component focusing on a single responsibility, 
+  enhancing maintainability and scalability.
+
+#### Alternatives Considered:
+
+- An embedded approach, where genre setting logic is part of a larger class managing all book attributes, was 
+  considered. However, this was rejected due to potential scalability issues and difficulty in maintaining code.
+
+![SetGenreSequenceDiagram.png](UML_diagrams/SetGenreSequenceDiagram.png)
+
+### BookList Class Component
+The `BookList` class is responsible for all actions involving the list of books that the user has.
+
+#### Overview
+
+The `BookList` class contains one protected static ArrayList named books. This ArrayList will contain Book objects. The methods in
+The `BookList` class contains one protected static ArrayList named books. This ArrayList will contain Book objects. 
+The methods in this class all change the ArrayList according to the command given.
+
+#### Detailed Workflow
+Apart from the constructor, the methods of this class like getSize(), addBook() all either return a piece of 
+information about the ArrayList,
+the book object that is selected or change an attribute of the ArrayList or selected book object. For the 
+printAllBooks() method, the ArrayList
+is iterated through, with the details of each book being printed out according to the toString() format of each book. 
+Other than that, methods like
+markDoneByIndex() and markUndoneByIndex() both will change the isRead() attribute of the book of the given index. This 
+class handles errors related to the
+ArrayList, throwing exceptions for invalid indexes and invalid actions based on current state (if trying to mark a book 
+that is already read).
+
+#### Implementation and Rationale
+
+### ParserMain Class Component
+The `ParserMain` class is responsible for parsing any input from the user and making sense of them to execute the correct commands.
+
+#### Overview
+By importing predefined string constants from `CommandList` class representing the valid commands, the `ParserMain` class 
+parses the input from a user using the `parseCommand` method. The class diagram below shows how `ParserMain` interacts with other
+classes.
+![ParserMain.png](UML_diagrams/ParserMain.png)
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/99406560/28e1f0be-6bd4-44a2-aa18-e152fef378f2)
Perhaps you could remove the icons in the class diagrams, as well as the attributes icons.